src/index.ts

```diff
@@ -18,136 +17,172 @@ interface Params {
+  throw new Error("We do not support this OS yet.");
 }
 
 // Detect binaries path
 // TODO: Consider installing Testkube in some random place, and add it to PATH environment variable
-const detectedPaths = (process.env.PATH || '').split(':').filter(Boolean).sort((a, b) => a.length - b.length);
-const writablePaths = (await Promise.all(detectedPaths.map(async dirPath => ({path: dirPath, writable: await fs.promises.access(dirPath, fs.constants.W_OK).then(() => true).catch(() => false)})))).filter(x => x.writable).map(x => x.path);
-const preferredPaths = ['/usr/local/bin', '/usr/bin'];
-const binaryDirPath = preferredPaths.find(x => writablePaths.includes(x)) || writablePaths[0];
-process.stdout.write(`Binary path: ${binaryDirPath || '<none>'}\n`);
+const detectedPaths = (process.env.PATH || "")
+  .split(":")
+  .filter(Boolean)
+  .sort((a, b) => a.length - b.length);
+const writablePaths = (
+  await Promise.all(
+    detectedPaths.map(async (dirPath) => ({
+      path: dirPath,
+      writable: await fs.promises
+        .access(dirPath, fs.constants.W_OK)
+        .then(() => true)
+        .catch(() => false),
+    }))
+  )
+)
+  .filter((x) => x.writable)
+  .map((x) => x.path);
+const preferredPaths = ["/usr/local/bin", "/usr/bin"];
+const binaryDirPath = preferredPaths.find((x) => writablePaths.includes(x)) || writablePaths[0];
+process.stdout.write(`Binary path: ${binaryDirPath || "<none>"}\n`);
 if (!binaryDirPath) {
-  throw new Error('Could not find a writable path that is exposed in PATH to put the binary.');
+  throw new Error("Could not find a writable path that is exposed in PATH to put the binary.");
 }
 
 // Detect if there is kubectl installed
-if (mode === 'kubectl') {
-  const hasKubectl = await which('kubectl', {nothrow: true});
-  process.stdout.write(`kubectl: ${hasKubectl ? 'detected' : 'not available'}.\n`);
+if (mode === "kubectl") {
+  const hasKubectl = await which("kubectl", { nothrow: true });
+  process.stdout.write(`kubectl: ${hasKubectl ? "detected" : "not available"}.\n`);
   if (!hasKubectl) {
-    throw new Error('You do not have kubectl installed. Most likely you need to configure your workflow to initialize connection with Kubernetes cluster.');
+    throw new Error(
+      "You do not have kubectl installed. Most likely you need to configure your workflow to initialize connection with Kubernetes cluster."
+    );
   }
 } else {
-  process.stdout.write('kubectl: ignored for Cloud integration\n');
+  process.stdout.write("kubectl: ignored for Cloud integration\n");
 }
 
-// Detect if there is Testkube CLI already installed
-if (await which('kubectl-testkube', {nothrow: true})) {
-  process.stdout.write('Looks like you already have the Testkube CLI installed. Skipping...\n');
+const existingTestkubePath = params.version ? toolCache.find("kubectl-testkube", params.version) : "";
+const isTestkubeInstalled =
+  existingTestkubePath.length > 0 || Boolean(await which("kubectl-testkube", { nothrow: true }));
```

Comment:
One issue though - shouldn't we have binary versioned too? Otherwise, changing `version` would have no effect, as the previously downloaded version will be used